src/js/carousel.js

```diff
@@ -0,0 +1,114 @@
+(function (window) {
+  function createDots() {
+    var existingDots = this.$carousel.getElementsByClassName('dots');
+    if (existingDots.length) {
+      this.$carousel.removeChild(existingDots[0]);
+    }
+    var dots = document.createElement('ul');
+    dots.className = 'dots';
+    var items = this.$carousel.getElementsByTagName('ul')[0];
+    this.$count = items.childElementCount;
+    for (var i = 0; i < items.childElementCount; i += 1) {
+      var dot = document.createElement('li');
+      dot.dataset.index = i;
+      dots.appendChild(dot);
+    }
+    this.$carousel.appendChild(dots);
+  }
+
+  function selectView(index) {
+    var scrollPosition = index * this.$width;
+    var items = this.$carousel.firstElementChild;
+    items.scrollTo(scrollPosition, 0);
+  }
+
+  function unselectAllDots() {
+    var dots = this.$carousel.lastElementChild;
+    for (var i = 0; i < dots.childElementCount; i += 1) {
```

Comment:
I started to use it because I think that makes sense what Douglas Crockford wrote in his book JavaScript - The Good Parts. I read the book, but I found on Oreilly an apendix:

http://archive.oreilly.com/pub/a/javascript/excerpts/javascript-good-parts/bad-parts.html 

Scroll down to the `++ --` subtitle.

Also I think it is more explicit the the `++` operator.